lib/portlayer/exec/container.go

```diff
@@ -393,16 +393,24 @@ func infraContainers(ctx context.Context, sess *session.Session) ([]mo.VirtualMa
 
 // find the childVM for this resource pool by name
 func childVM(ctx context.Context, sess *session.Session, name string) (*vm.VirtualMachine, error) {
-	searchIndex := object.NewSearchIndex(sess.Client.Client)
-	child, err := searchIndex.FindChild(ctx, VCHConfig.ResourcePool.Reference(), name)
```

Comment:
Wasn't aware of that issue with vSAN, but not surprised I guess.  Another possible option, if we can use the ID to set the vm Config.InstanceUuid, then use SearchIndex.FindByUuid

We'd may need to convert the ID to uuid using for example: https://godoc.org/github.com/google/uuid#NewSHA1